aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -243,6 +247,170 @@ std::tuple<Tensor, Tensor> prelu_backward_cuda(const Tensor& grad_out_, const Te
   return std::tuple<Tensor, Tensor>{input_grad, weight_grad};
 }
 
+// -----------------------------------
+// rrelu
+// -----------------------------------
+template <typename scalar_t, int unroll_factor, typename F>
+#if __CUDA_ARCH__ >= 350 || defined __HIP_PLATFORM_HCC__
+C10_LAUNCH_BOUNDS_2(256, 4)
+#endif
+__global__ void rrelu_with_noise_cuda_kernel(
+    int numel,
+    PhiloxCudaState philox_args,
+    scalar_t* output,
+    scalar_t* input,
+    scalar_t* noise,
+    double lower,
+    double upper,
+    const F& random_func) {
+  auto seeds = at::cuda::philox::unpack(philox_args);
+  int idx = blockIdx.x * blockDim.x + threadIdx.x;
+  curandStatePhilox4_32_10_t state;
+  curand_init(std::get<0>(seeds),
+              idx,
+              std::get<1>(seeds),
+              &state);
+
+  int grid_stride = blockDim.x * gridDim.x * unroll_factor;
+  int rounded_size = ((numel - 1) / grid_stride + 1) * grid_stride;
+  double range = upper - lower;
+
+  for (int linear_index = idx; linear_index < rounded_size; linear_index += grid_stride) {
+    auto rand = random_func(&state);
```

Comment:
Can you please add static assert here that sizeof(rand)/sizeof(rand.x) == unroll_factor? Otherwise your (&rand.x)[ii] access is unsafe.